packages/flutter/lib/src/painting/placeholder_span.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui show ParagraphBuilder, PlaceholderAlignment;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/services.dart';
+
+import 'basic_types.dart';
+import 'inline_span.dart';
+import 'text_style.dart';
+import 'text_painter.dart';
+import 'text_span.dart';
+
+/// An immutable placeholder that is embedded inline within text.
+///
+/// [PlaceholderSpan] represents an abstract generic placeholder. [WidgetSpan] should be
+/// used instead to specify the contents of the placeholder. A [PlaceholderSpan]
+/// by itself does not contain useful information to change a TextSpan.
+///
+/// [PlaceholderSpan]s must be leaf nodes in the [TextSpan] tree. [PlaceholderSpan]s
+/// cannot have any [TextSpan] children.
+///
+/// [PlaceholderSpan]s will be ignored when passed into a [TextPainter] directly.
+/// A [WidgetSpan] should be passed into a [RichText] widget instead.
```

Comment:
this paragraph is dubious. PlaceholderSpans can't be passed directly, since they can't be constructed. WidgetSpans are PlaceholderSpans, and they certainly won't be ignored.

Saying that WidgetSpan should be passed to RichText is weird at this level of the API; it's not the docs for WidgetSpan, and it's at a layer where RichText might not be available (and in any case Text is probably a better widget to be using this with).